pyvista/core/_validation/validate.py

```diff
@@ -708,7 +721,10 @@ def _array_from_vtkmatrix(
     return array
 
 
-def validate_number(num, /, *, reshape: bool = True, **kwargs):
+_T = TypeVar('_T', float, int)
+
+
+def validate_number(num: _T, /, *, reshape: bool = True, **kwargs) -> _T:
```

Comment:
Sphinx raises warnng that we cannot reference `_T` in document.
```suggestion
def validate_number(num: float | int, /, *, reshape: bool = True, **kwargs) -> float | int:
```

Reply:
Hmm. This is because of #6895.  TypeVar is better because it handles the `int -> int` mapping but we can use overloads instead (and add these in a separate PR)